src/background/services/storage.ts

```diff
@@ -82,6 +81,24 @@ export class StorageService {
     return { ...data, url }
   }
 
+  async setPartial<TKey extends Partial<StorageKey>>(
+    data: Partial<{
+      [K in TKey]: Storage[TKey]
+    }>
+  ): Promise<void> {
+    const storageData = await this.get([
+      'enabled',
+      'connected',
+      'amount',
+      'rateOfPay',
+      'minRateOfPay',
+      'maxRateOfPay',
+      'walletAddress',
+      'publicKey'
+    ])
+    await this.browser.storage.local.set({ ...storageData, ...data })
+  }
+
```

Comment:
This is unrelated now, right? Can be removed?

Reply:
Currently not used, but I thought it's a good util to be used in the feature if we want.

Reply:
I tend to keep PRs as focused as they can be, but alright.